lib/platform/platform.c

```diff
@@ -401,7 +417,30 @@ void gas_write64(struct switchtec_dev *dev, uint64_t val, uint64_t __gas *addr)
 void memcpy_to_gas(struct switchtec_dev *dev, void __gas *dest,
 		   const void *src, size_t n)
 {
-	dev->ops->memcpy_to_gas(dev, dest, src, n);
+
+	uint32_t gas_addr;
+	int ret;
+
+	gas_addr = (uint32_t)(dest - (void __gas *)dev->gas_map);
+
+	if (gas_addr < MRPC_MAX_DATA_LEN)
```

Comment:
Per fw implementation, for a remote MRPC GAS access, the remote MRPC region in GAS is also involved in the execution. So there's no way to remote access the MRPC region.

It's a bug here, the limitation should be 4KB of the whole MRPC region instead of the max data len. I will ask firmware to return error on request to unaccessible regions.